pinot-common/src/test/java/org/apache/pinot/sql/parsers/CalciteSqlCompilerTest.java

```diff
@@ -3072,6 +3072,16 @@ public void testExtractTableNamesFromNode() {
     Assert.assertEquals(tableNames.get(0), "tbl1");
     Assert.assertEquals(tableNames.get(1), "tbl2");
 
+    // query with UNION clause
+    query = "SELECT * FROM tbl1 UNION ALL SELECT * FROM tbl2 UNION ALL SELECT * FROM tbl3";
```

Comment:
can we add several more for parsing? for example
`WITH tmpTable AS (...) SELECT * FROM tbl1 UNION ALL tmpTable`
^ here tmpTable should not be part of the check

Reply:
Added the table alias examples.
Somehow SetOps require an expression instead of a table name, so need to change to
```
WITH tmpTable AS (...) SELECT * FROM tbl1 UNION ALL SELECT * FROM tmpTable
```